templates/default/docker.yaml.erb

```diff
@@ -0,0 +1,50 @@
+# Generated by Chef, local modifications will be overwritten
+
+init_config:
+  <% if @init_config %>
+  <% if @init_config.key?('docker_root') -%>
+  docker_root: <%= @init_config['docker_root'] %>
+  <% end -%>
+  <% if @init_config.key?('socket_timeout') -%>
+  socket_timeout: <%= @init_config['socket_timeout'] %>
+  <% end -%>
+  <% end -%>
+
+instances:
+  <% @instances.each do |i| -%>
```

Comment:
This is cool, but I think I tried to explain that we should try to use the more flexible method as seen in newer examples like [this one](https://github.com/DataDog/chef-datadog/blob/master/templates/default/http_check.yaml.erb) where the `instances` hash is rendered directly from attributes structure, instead of us needing to change & release whenever an Agent check adds/changes a config flag.

I've been trying to think how this might work for init_config as well, have not figured that out yet.